internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,407 @@
+	"http://ip.tyk.nu",
+}
+
+func (opts *Opts) newWhitelist() *atlas.ProjectIPWhitelist {
+	return &atlas.ProjectIPWhitelist{
+		GroupID:   opts.ConfigProjectID(),
+		Comment:   accessListComment,
+		IPAddress: opts.ipAddress,
+	}
+}
+
+func (opts *Opts) newCluster() *atlas.Cluster {
+	diskSizeGB := float64(diskSizeGB)
+	return &atlas.Cluster{
+		GroupID:             opts.ConfigProjectID(),
+		ClusterType:         replicaSet,
+		ReplicationSpecs:    []atlas.ReplicationSpec{opts.newReplicationSpec()},
+		ProviderSettings:    opts.newProviderSettings(),
+		MongoDBMajorVersion: mdbVersion,
+		DiskSizeGB:          &diskSizeGB,
+		Name:                opts.clusterName,
+	}
+}
+
+func (opts *Opts) newReplicationSpec() atlas.ReplicationSpec {
+	var (
+		readOnlyNodes int64 = 0
+		priority      int64 = 7
+		shards        int64 = shards
+		members       int64 = members
+	)
+	replicationSpec := atlas.ReplicationSpec{
+		NumShards: &shards,
+		ZoneName:  zoneName,
+		RegionsConfig: map[string]atlas.RegionsConfig{
+			opts.region: {
+				ReadOnlyNodes:  &readOnlyNodes,
+				ElectableNodes: &members,
+				Priority:       &priority,
+			},
+		},
+	}
+	return replicationSpec
+}
+
+func (opts *Opts) newProviderSettings() *atlas.ProviderSettings {
+	return &atlas.ProviderSettings{
+		InstanceSizeName: tier,
+		ProviderName:     opts.provider,
+		RegionName:       opts.region,
+	}
+}
+
+// askRequiredFlags allows the user to set required flags by using interactive prompts
+func (opts *Opts) askRequiredFlags() error {
+	if opts.dbUsername == "" {
+		dbUsername := ""
+		dbUsernamePrompt := &survey.Input{
+			Message: "Insert the Username for authenticating to MongoDB [Press Enter to use an autogenerated username]",
+			Help:    usage.DBUsername,
+		}
+
+		if err := survey.AskOne(dbUsernamePrompt, &dbUsername); err != nil {
+			return err
+		}
+		opts.dbUsername = dbUsername
+
+		if opts.dbUsername == "" {
+			n, err := e2e.RandInt(maxRandNum)
+			if err != nil {
+				return err
+			}
+			opts.dbUsername = "quickStart_" + n.String()
+		}
+	}
+
+	if opts.dbUserPassword == "" {
+		dbPassword := ""
+		dbPasswordPrompt := &survey.Password{
+			Message: "Insert the Password for authenticating to MongoDB [Press Enter to use an autogenerated password]",
+			Help:    usage.Password,
+		}
+
+		if err := survey.AskOne(dbPasswordPrompt, &dbPassword); err != nil {
+			return err
+		}
+
+		opts.dbUserPassword = dbPassword
+		if opts.dbUserPassword == "" {
+			p, err := newAutogeneratedPassword(passwordLength)
+			if err != nil {
+				return err
+			}
+			opts.dbUserPassword = p
+		}
+	}
+
+	if opts.ipAddress == "" {
+		answer := ""
+		publicIP, err := newIPAddress()
+		message := "Insert the IP entry to add to the Access List"
+		if err == nil {
+			message = fmt.Sprintf(`Insert the IP entry to add to the Access List [Press Enter to use your public IP "%s"]`, publicIP)
+		}
+
+		publicIPPrompt := survey.Input{
+			Message: message,
+			Help:    usage.AccessListIPEntry,
+		}
```

Comment:
you can set a default for a prompt and it will show like this
```bash
mongocli config --service cloud-manager -P broken
You are configuring a profile for mongocli.

All values are optional and you can use environment variables (MCLI_*) instead.

Enter [?] on any option to get help.

? Public API Key: [? for help] (JQVAXGIA)
```

that value between parenthesis is the value already set